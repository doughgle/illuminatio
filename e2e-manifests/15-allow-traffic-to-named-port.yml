apiVersion: v1
kind: Namespace
metadata:
  name: 15-allow-traffic-to-named-port
  labels:
    illuminatio-e2e: 15-allow-traffic-to-named-port
---
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: 15-allow-traffic-to-named-port
  labels:
    app: myfancydeployment
    role: api
    illuminatio-e2e: 15-allow-traffic-to-named-port
  name: my-deployment
spec:
  replicas: 1
  revisionHistoryLimit: 2
  selector:
    matchLabels:
      app: myfancydeployment
      role: api
  template:
    metadata:
      labels:
        app: myfancydeployment
        role: api
    spec:
      containers:
      - image: nginx
        imagePullPolicy: Always
        name: myfancydeployment
        ports:
        - containerPort: 80
          name: mynamedport
          protocol: TCP
---
apiVersion: v1
kind: Service
metadata:
  namespace: 15-allow-traffic-to-named-port
  labels:
    app: myfancydeployment
    role: api
    illuminatio-e2e: 15-allow-traffic-to-named-port
  name: my-deployment-svc
spec:
  ports:
  - port: 80
    protocol: TCP
    targetPort: mynamedport
  selector:
    app: myfancydeployment
    role: api
  type: ClusterIP
---
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  namespace: 15-allow-traffic-to-named-port
  name: api-allow
  labels:
    illuminatio-e2e: 15-allow-traffic-to-named-port
spec:
  podSelector:
    matchLabels:
      app: myfancydeployment
      role: api
  ingress:
  - from:
      - podSelector:
          matchLabels:
            role: frontend
    ports:
      - port: mynamedport
        protocol: TCP
