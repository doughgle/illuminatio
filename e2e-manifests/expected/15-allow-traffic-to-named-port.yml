15-allow-traffic-to-named-port:illuminatio-inverted-role=frontend:
  15-allow-traffic-to-named-port:app=myfancydeployment,role=api:
    -mynamedport:
      success: true
15-allow-traffic-to-named-port:role=frontend:
  15-allow-traffic-to-named-port:app=myfancydeployment,role=api:
    mynamedport:
      success: true
illuminatio-inverted-15-allow-traffic-to-named-port:illuminatio-inverted-role=frontend:
  15-allow-traffic-to-named-port:app=myfancydeployment,role=api:
    -mynamedport:
      success: true
illuminatio-inverted-15-allow-traffic-to-named-port:role=frontend:
  15-allow-traffic-to-named-port:app=myfancydeployment,role=api:
    -mynamedport:
      success: true
